src/screencast/wlr_screencast.c

```diff
@@ -1,31 +1,72 @@
 #define _POSIX_C_SOURCE 200809L
 
 #include "wlr_screencast.h"
+
+#include "wlr-screencopy-unstable-v1-client-protocol.h"
+#include "xdg-output-unstable-v1-client-protocol.h"
+#include <fcntl.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/param.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <wayland-client-protocol.h>
+
+#include "screencast.h"
+#include "pipewire_screencast.h"
 #include "xdpw.h"
+#include "logger.h"
 
-void wlr_frame_free(struct xdpw_state *state) {
-	zwlr_screencopy_frame_v1_destroy(state->screencast.wlr_frame);
-	munmap(state->screencast.simple_frame.data, state->screencast.simple_frame.size);
-	wl_buffer_destroy(state->screencast.simple_frame.buffer);
+void xdpw_wlr_frame_free(struct xdpw_screencast_instance *cast) {
+	zwlr_screencopy_frame_v1_destroy(cast->wlr_frame);
+	munmap(cast->simple_frame.data, cast->simple_frame.size);
```

Comment:
We should reset all these free'ed fields to NULL/zero, to make sure we don't use-after-free and make debugging easier.